src/ClusterBootstrap/cloud_init_aztools.py

```diff
@@ -334,6 +334,9 @@ def add_machine_in_parallel(cmds, args):
 
 
 def add_machine(vmname, spec, verbose, dryrun, output_file):
+    multual_exclusive_roles = set(["infra", "worker", "elasticsearch", "mysqlserver"])
+    mul_ex_role_in_spec = list(set(spec["role"]) & multual_exclusive_roles)
+    assert len(mul_ex_role_in_spec) <= 1 and "We don't allow role overlapping between these roles."
```

Comment:
Can we use `assert <condition>, <msg>` instead of `assert <condition> and <msg>`? Please include what roles are not allowed overlapping in the message as well.

Reply:
corrected.